lambda/matching.ml

```diff
@@ -2722,6 +2717,65 @@ let mk_failaction_pos partial seen ctx defs =
         | Some lam -> string_of_lam lam
         );
     (fail, [], jumps)
+  ) else (
+    (* Compute a specialized context for each failure pattern. *)
+    let fail_pats =
+      List.map (fun pat -> (pat, Context.lub pat ctx)) fail_pats in
+    (* Determine which exit is appropriate for which failure patterns. *)
+    let rec exit_failpats defs fail_pats acc =
+      match Default_environment.pop defs with
+      | Some ((idef, pss), rem) ->
+          (* Collect the patterns whose context matches the
+             matrix [pss] of the exit [idef]. *)
+          let now, later =
+            List.partition_map (fun (p, p_ctx) ->
+              if Context.matches p_ctx pss
+              then Either.Left p
+              else Either.Right (p, ctx)
```

Comment:
Shouldn't this be `p_ctx` instead of `ctx` ?

Reply:
It should, indeed, thanks! (One can check the previous behaviour of the `List.partition` call.) I rewrote this with an `as` pattern to be safer.